veracruz-utils/src/ecdsa.rs

```diff
@@ -0,0 +1,70 @@
+//! ECDSA functions.
+//!
+//! ## Authors
+//!
+//! The Veracruz Development Team.
+//!
+//! ## Licensing and copyright notice
+//!
+//! See the `LICENSE_MIT.markdown` file in the Veracruz root directory
+//! for information on licensing and copyright.
+
+use crate::der;
+use crate::sha256::sha256;
+use psa_crypto;
+use psa_crypto::operations::asym_signature;
+use psa_crypto::operations::key_management;
+use psa_crypto::types::algorithm::{AsymmetricSignature, Hash};
+use psa_crypto::types::key::{Attributes, EccFamily, Lifetime, Policy, Type, UsageFlags};
+
+/// Attributes for exportable ECDSA key for signing.
+fn attributes() -> Attributes {
+    let mut usage_flags: UsageFlags = Default::default();
+    usage_flags.set_sign_hash().set_export();
+    Attributes {
+        // ECDST prime256v1 = secp256r1 = P-256
+        key_type: Type::EccKeyPair {
+            curve_family: EccFamily::SecpR1,
+        },
+        bits: 256,
+        lifetime: Lifetime::Volatile,
+        policy: Policy {
+            usage_flags,
+            permitted_algorithms: AsymmetricSignature::Ecdsa {
+                hash_alg: Hash::Sha256.into(),
+            }
+            .into(),
+        },
+    }
+}
+
+/// Generate ECDSA keypair (public, private) in DER encoding.
+pub fn generate() -> (Vec<u8>, Vec<u8>) {
+    psa_crypto::init().unwrap();
```

Comment:
We should avoid `unwrap()`.  When would this fail?  (And all other uses of `unwrap()` in this file...). Can we not make these functions return an error type?